jgrapht-core/src/main/java/org/jgrapht/alg/flow/EdmondsKarpMaximumFlow.java

```diff
@@ -144,64 +141,67 @@ public EdmondsKarpMaximumFlow(DirectedGraph<V, E> network, double epsilon)
      */
     public MaximumFlow<E> buildMaximumFlow(V source, V sink)
     {
+        this.calculateMaximumFlow(source, sink);
+        maxFlow = composeFlow();
+        return new MaximumFlowImpl<>(maxFlowValue, maxFlow);
+    }
+
+    /**
+     * Sets current source to <tt>source</tt>, current sink to <tt>sink</tt>,
+     * then calculates maximum flow from <tt>source</tt> to <tt>sink</tt>. Note,
+     * that <tt>source</tt> and <tt>sink</tt> must be vertices of the <tt>
+     * network</tt> passed to the constructor, and they must be different.
+     * If desired, a flow map can be queried afterwards; this will not require
+     * a new invocation of the algorithm.
+     *
+     * @param source source vertex
+     * @param sink sink vertex
+     */
+    public double calculateMaximumFlow(V source,V sink){
         super.init(vertexExtensionsFactory, edgeExtensionsFactory);
 
         if (!network.containsVertex(source)) {
             throw new IllegalArgumentException(
-                "invalid source (null or not from this network)");
+                    "invalid source (null or not from this network)");
         }
         if (!network.containsVertex(sink)) {
             throw new IllegalArgumentException(
-                "invalid sink (null or not from this network)");
+                    "invalid sink (null or not from this network)");
         }
 
         if (source.equals(sink)) {
             throw new IllegalArgumentException("source is equal to sink");
         }
 
-        currentSource = extendedVertex(source);
-        currentSink = extendedVertex(sink);
-
-        Map<E, Double> maxFlow;
-
-        double maxFlowValue;
+        currentSource = getVertexExtension(source);
+        currentSink = getVertexExtension(sink);
 
         for (;;) {
             breadthFirstSearch();
 
             if (!currentSink.visited) {
-                maxFlow = composeFlow();
-                maxFlowValue = 0.0;
-                for (E e : network.incomingEdgesOf(currentSink.prototype)) {
-                    maxFlowValue += maxFlow.get(e);
-                }
                 break;
             }
 
-            augmentFlow();
+            maxFlowValue+=augmentFlow();
```

Comment:
I honestly can't spot the difference compared to what you for example wrote: "reverseEdge.capacity += flowDifference"
I can't hardly imagine you are referring to the space before and after the +=?